plugins/CorePluginsAdmin/Controller.php

```diff
@@ -32,57 +35,38 @@ class Controller extends Plugin\ControllerAdmin
     const DEACTIVATE_NONCE = 'CorePluginsAdmin.deactivatePlugin';
     const UNINSTALL_NONCE = 'CorePluginsAdmin.uninstallPlugin';
 
-    private $validSortMethods = array('popular', 'newest', 'alpha');
-    private $defaultSortMethod = 'popular';
-
     /**
      * @var Translator
      */
     private $translator;
 
-    public function __construct(Translator $translator)
-    {
-        $this->translator = $translator;
-
-        parent::__construct();
-    }
-
-    public function marketplace()
-    {
-        self::dieIfMarketplaceIsDisabled();
-
-        $show = Common::getRequestVar('show', 'plugins', 'string');
-        $query = Common::getRequestVar('query', '', 'string', $_POST);
-        $sort = Common::getRequestVar('sort', $this->defaultSortMethod, 'string');
-        if (!in_array($sort, $this->validSortMethods)) {
-            $sort = $this->defaultSortMethod;
-        }
-        $mode = Common::getRequestVar('mode', 'admin', 'string');
-        if (!in_array($mode, array('user', 'admin'))) {
-            $mode = 'admin';
-        }
+    /**
+     * @var PluginInstaller
+     */
+    private $pluginInstaller;
 
-        $view = $this->configureView('@CorePluginsAdmin/marketplace');
+    /**
+     * @var Plugins
+     */
+    private $marketplacePlugins;
 
-        $marketplace = new Marketplace();
+    /**
+     * @var Plugin\Manager
+     */
+    private $pluginManager;
 
-        $showThemes = ($show === 'themes');
-        $view->plugins = $marketplace->searchPlugins($query, $sort, $showThemes);
-        $view->showThemes = $showThemes;
-        $view->mode = $mode;
-        $view->query = $query;
-        $view->sort = $sort;
-        $view->installNonce = Nonce::getNonce(static::INSTALL_NONCE);
-        $view->updateNonce = Nonce::getNonce(static::UPDATE_NONCE);
-        $view->isSuperUser = Piwik::hasUserSuperUserAccess();
+    public function __construct(Translator $translator, PluginInstaller $pluginInstaller, Plugins $marketplacePlugins)
+    {
+        $this->translator = $translator;
+        $this->pluginInstaller = $pluginInstaller;
+        $this->marketplacePlugins = $marketplacePlugins;
+        $this->pluginManager = Plugin\Manager::getInstance();
 
-        return $view->render();
+        parent::__construct();
     }
 
     private function createUpdateOrInstallView($template, $nonceName)
     {
```

Comment:
This dieIfMarketplace... is not needed anymore since it is in the Marketplace directly and if plugin was disabled it is protected from execution by the framework